mdp/signal_node.py

```diff
@@ -173,10 +174,15 @@ def _function_infodict(func):
             argnames.append(varargs)
         if varkwargs:
             argnames.append(varkwargs)
-        signature = inspect.formatargspec(regargs,
-                                          varargs,
-                                          varkwargs,
-                                          defaults,
+
+        # python 2/3 compatibility
+        try:
+            signature = _legacysignature(inspect.signature(func))
+        except AttributeError:
+            signature = inspect.formatargspec(regargs,	      
+                                          varargs,	
+                                          varkwargs,	
+                                          defaults,	
```

Comment:
Either fix the alignment or put it like this:
```
signature = inspect.formatargspec(
        regargs, varargs, varkwargs, defaults, formatvalue=lambda value: "")[1:-1]
```
(additional linebreak as needed)

Reply:
The alignment should not be different, as then the column limit is exceeded by the last argument.

Reply:
Please make it a three-liner then. This non-alignment just looks wrong.
```
signature = inspect.formatargspec(
        regargs, varargs, varkwargs, defaults,
        formatvalue=lambda value: "")[1:-1]
```